### PR TITLE
octopus rgw: under fips, set flag to allow md5 in select rgw ops

### PR DESCRIPTION
**Pull request by mkogan1**
backport tracker: https://tracker.ceph.com/issues/53008

backport of: #43503
parent tracker: https://tracker.ceph.com/issues/52900

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/rgw_tools.h [hunk 1]
```diff
@@ -111,7 +111,12 @@ class RGWEtag
   H hash;
 
 public:
-  RGWEtag() {}
+  RGWEtag() {
+    if constexpr (std::is_same_v<H, MD5>) {
```
Comment (mattbenjamin) [inline src/rgw/rgw_tools.h:115]: this is very pretty c++, I didn't notice this in the earlier change

**Review by mattbenjamin — APPROVED** [review]
this looks good to me--I don't know if it covers all cases, but we have reason to believe it does

**Comment by mkogan1** [comment]
jenkins test api

**Comment by mkogan1** [comment]
jenkins test api

**Comment by noskovao** [comment]
Can we get move approvals?
We have the same issue with RGW + fips

**Comment by prazumovsky** [comment]
Hello! Do we have any updates on this backport? We're waiting for it and still no new states here :)

**Comment by prazumovsky** [comment]
Hello! Is it possible to make another octopus release which will include this backport in the near time?

**Comment by Jiawei0227** [comment]
+1 can we make a new release with this change?

**Comment by Jiawei0227** [comment]
Gentle ping, what's the cadence of the release to v15 right now? This is blocking the usage of object storage for us

**Comment by prazumovsky** [comment]
Gentle reminder that we still waiting ceph octopus bump with this fix because it blocks our FIPS-enabled deployment.

**Pull request merged**